### ISSUE TITLE
Accessibility: Font-sizes and other measurements

### ISSUE THREAD

**Issue by StfBauer** [OfficeDev/Office-UI-Fabric#528]
Currently all font sizes have been implemented to use pixel values, I guess there are some other pixel based values too.
The problem is that if a user choose to use a larger text size for better reading Office UI Fabric won't scale properly.
I think it would be better to use em, rem or % based on the default text size for all measurements.

I know this would have a huge impact on currenty Office UI Fabric but would be a benefit for accessibility.

**Comment by mikewheaton** [comment]
Our designers work in pixels, so it would be difficult change to another unit without introducing communication issues. We've discussed `em` in the past but it presents issues with text size inheritance, and unfortunately `rem` does not have full support on IE9. My understanding is that with modern browsers (even IE9!) providing page zooming, pixels are now considered acceptable.

I've flagged this as needs-discussion so that we can go into more depth with the whole team.

**Comment by StfBauer** [comment]
As always thank you for your feedback. In additon I have a couple of remarks to your feedback.

I know it is a common issue when graphic designer define how web design have to be implemented by web designer/web developer.
In addition with a proper viewport meta tag and iPhone would render 16px as 16pixels which is pretty tiny on devices with higher pixel density. A thing not covered by Office UI Fabric.
To be honest the calculation from pixel to em is not that complicated. Ever browser has 16px as the default text size define. Base on this you can easily calculate relative values. Especially when you use SASS.
Pixel values on the other hand are against the web accessibility guide lines too: see http://www.w3.org/WAI/GL/WCAG20-TECHS/css.html#C14

You really tend to support IE9. This leaves me a bit confused because Microsoft official droped the support for browser versions older than IE11. The support have been dropped on 12th of January this year. So why do you still support such old browser?

In addition Office 365 only supports the latest browser see:
https://products.office.com/en/office-system-requirements
I guess this won't be tested with Office 365 Products such as SharePoint, Outlook and so on. I'm sure you will get more information on that internally.

Regarding to Zooming: This works well when you have a full browser window but not when use Office UI Fabric in an Office Add-in. There is no way to trigger zoom. In this case you need to add a control that allows the user to resize the text. A thing that won't be used for ages now.

**Comment by mikewheaton** [comment]
Good points about calculating ems from pixel values and the potential problems with trying to control font sizes in Office Add-ins. We can certainly revisit this discussion as it's been a while since we last considered a change here.

Browser support is a complicated situation. We'd love to drop IE9 (and IE10) support both to reduce the amount of browser testing we have to do and to gain access to new features like flexbox. However, Fabric is in use on some products that are seeing more significant IE9 usage than the global stats would suggest. Some of the nicer touches like animation can be unsupported on IE9 (our support matrix allows for a "diminished experience") but all of the core styles must work.

**Comment by StfBauer** [comment]
Thank you. 👍   Wasn't aware of the dimished experience.
Let's hope this support will be dropped soon.

**Comment by ericthompson** [comment]
@StfBauer - we totally hear you. As @mikewheaton said, we're still seeing some usage (last time we checked). We'll kick off the discussions on our end and report back if there are any updates.

**Comment by StfBauer** [comment]
Thanks. The problem I see is as long as you support it there will be a use for it. Even when other surrounding systems already dropped the support for that.
In case of the current 'dimished experience' it is understandable.

**Comment by mikewheaton** [comment]
Closing this issue, although I'm sure we'll be discussing it again before long. 😄

**Comment by jordandobson** [comment]
IE9 does actually support REM which is essentially the new standard now a
days. :)

We shouldn't be using fixed values like px. Stefan is completely right. The
translation is easy peasy. ;)
